source/hp/dof_handler.cc

```diff
@@ -2022,11 +2040,6 @@ namespace hp
                 saved_subdomain_ids[index] = cell->subdomain_id();
                 cell->set_subdomain_id(true_subdomain_ids[index]);
               }
-
-            // Make sure every processor knows the active_fe_indices
-            // on both its own cells and all ghost cells.
-            dealii::internal::hp::DoFHandlerImplementation::Implementation::
-              communicate_active_fe_indices(*this);
```

Comment:
This call is not mandatory, but we could leave it in to ensure that all active fe indices are set correctly.

Reply:
What happens if you take it out? Are there ghost cells that then have the wrong active fe index?

Reply:
We can safely leave it out. It was initially meant to make sure that the indices are indeed set correctly, but I cannot think of a case in which they are not.

Thinking about it again, I realize that working on non-artificial cells is sufficient, thus we do not need to store and recover the true subdomain ids. I'll give it a try tomorrow and check if everything is fine.